src/realm/object-store/impl/deep_change_checker.cpp

```diff
@@ -397,7 +397,8 @@ void CollectionKeyPathChangeChecker::find_changed_columns(std::vector<ColKey>& c
         auto last_column_key = last_key_path_element.second;
         if (last_column_key.get_type() == col_type_BackLink) {
             auto iterator = m_info.tables.find(table.get_key());
-            if (iterator != m_info.tables.end() && !iterator->second.insertions_empty()) {
+            if (iterator != m_info.tables.end() &&
+                (!iterator->second.insertions_empty() || !iterator->second.modifications_empty())) {
```

Comment:
We likely need to check deletions here as well. See https://github.com/realm/realm-swift/issues/7684

Reply:
This should have been resolved now.